test/test_sanity.py

```diff
@@ -817,3 +817,28 @@ def test_bootstrap(self):
 
     # Now the compiler should work again
     self.run_process([EMCC, test_file('hello_world.c')])
+
+  # Verify that running bootstrap.py in a first-time run scenario should not
+  # cause an exception. (A first time run scenario is before .emscripten, env.
+  # vars nor PATH has been configured)
+  def test_bootstrap_without_em_config(self):
```

Comment:
Can you just do `@with_env_modify({'EM_CONFIG': "/path/does/not/exist"})` maybe?

Reply:
Sure - folded that to the other env. modify code.

Reply:
But can you literally just write this test as:

```
@with_env_modify({'EM_CONFIG': "/path/does/not/exist"})
 def test_bootstrap_without_em_config(self):
   # Verify that bootstrap works even without a valid config file.
   self.run_process([shared.bat_suffix(shared.path_from_root('bootstrap'))])
```

Reply:
Oh I see, the behavior of `from tools import shared` is different with

```
@with_env_modify({'EM_CONFIG': "/path/does/not/exist"})
```
versus
```
@with_env_modify({'EM_CONFIG': None})
```

It is better to test the scenario that first-time emsdk install also exercises, when there are no env. vars yet set. This way the test matches what happens on a new installation, and won't rely on EM_CONFIG pointing to a nonexistent file (emsdk doesn't do that either on first-time install)

Reply:
Fair enough.  I think its probably overkill but I get that there could be difference.